core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -176,4 +176,20 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
+   * algorithm, the call will fail.  This method may only be called once, and only before
+   * {@link #sendHeaders}.  The compressor to use will be looked up in the
+   * {@link CompressorRegistry}.  Default gRPC servers support the "gzip" compressor.
+   *
+   * @param compressor the name of the compressor to use.
+   * @throws IllegalArgumentException if the compressor name can not be found.
+   * @throws UnsupportedOperationException if the implementation does not support setting
```

Comment:
Isn't throwing painful for users? It seems you expect it to almost never be thrown, but yet users are sort of expected to handle the condition. It doesn't force implementations to override the method, instead it forces clients to handle it.

Could it please be a noop, or maybe return a boolean?